impeller/display_list/dl_dispatcher.cc

```diff
@@ -623,7 +623,7 @@ void DlDispatcherBase::saveLayer(const SkRect& bounds,
                      ? ContentBoundsPromise::kMayClipContents
                      : ContentBoundsPromise::kContainsContents;
   std::optional<Rect> impeller_bounds;
-  if (!options.content_is_unbounded()) {
+  if (!options.content_is_unbounded() || options.bounds_from_caller()) {
```

Comment:
found another bug. even if contents are unbounded, we should still treat them as bounded if the caller specified bounds